contrib/core-contract-tests/tests/bns/name_register.test.ts

```diff
@@ -0,0 +1,597 @@
+import { Cl } from "@stacks/transactions";
+import { beforeEach, describe, expect, it } from "vitest";
+import { createHash } from "node:crypto";
+
+const accounts = simnet.getAccounts();
+const alice = accounts.get("wallet_1")!;
+const bob = accounts.get("wallet_2")!;
+const charlie = accounts.get("wallet_3")!;
+
+const cases = [
+  {
```

Comment:
@hugocaillard, this may be a good candidate for a fast-check generator (in a future PR) 🤔

Reply:
Maybe in a future PR. We could consider it as part of https://github.com/stacks-network/stacks-blockchain/issues/4030